plugin.audio.radiobrowser/main.py

```diff
@@ -196,37 +262,40 @@ def delFromMyStations(stationid):
         value = base64.b32decode(args['value'][0])
         value = value.decode('utf-8')
         param = dict({key:value})
+        param['order'] = 'clickcount'
+        param['reverse'] = True
 
-    data = downloadFile(url, param)
+    data = downloadApiFile(url, param)
     addPlayableLink(data)
     xbmcplugin.endOfDirectory(addon_handle)
 
 elif mode[0] == 'play':
-    stationid = args['stationid'][0]
-    data = downloadFile('http://www.radio-browser.info/webservice/v2/json/url/'+str(stationid),None)
+    stationuuid = args['stationuuid'][0]
+    data = downloadApiFile('/json/url/' + str(stationuuid),None)
     dataDecoded = json.loads(data)
     uri = dataDecoded['url']
+    xbmc.Player().stop()
     xbmc.Player().play(uri)
```

Comment:
Plugins should not call Player.play() but use xbmcplugin.setResolvedUrl instead. Can you adjust the add-on?

Reply:
yes i will immediatelly

Reply:
it does not work :(
maybe the wrong place to ask, but do you know of any reason why the addon handle can get the value of "-1" ?
addon_handle = int(sys.argv[1])

Reply:
isFolder=True does make the handle transfer work again.... that is why it works for all other pages but the deepest one where the real streams are...

Reply:
i found the solution, i was missing
li.setProperty('IsPlayable', 'true')

Reply:
Yeah sorry for the delay. You need to set the IsPlayable property to true to make it work. The reason behind this request is that, if you don't do it, everytime you play a radio channel a new entry is added to the database. Plugin setresolvedurl assigns a unique path to each of the items, hence, you're always playing the same file although the resolved stream url might difer